<!DOCTYPE html>
<html lang="en">
<head>
  <meta charset="UTF-8">
  <meta http-equiv="X-UA-Compatible" content="IE=edge">
  <meta name="viewport" content="width=device-width, initial-scale=1.0">
  <title>Document</title>
</head>
<body>
  <script>
    new Promise((resolve,reject) => {
      // resolve(1)
      reject(1)
    }).then(
      value => {
        console.log('onResolved1()',value);
        return 2
      },
      // 这里没有写失败的回调函数，相当于写了: reason => {throw reason} 代码
    ).then(
      value => {
        console.log('onResolved2()',value);
        return 3
      }
    ).then(
      value => {
        console.log('onRejected3()',value);
      }
    ).catch(reason => {
      console.log('onRejected1()',reason);  // onRejected1() 1
      // 下面的代码 -> 返回一个 pending 的 Promise，目的就是使用 Promise 的 then 链式调用时，在中间中断，不再调用后面的回调函数
      return new Promise(() => {})  
    }).then(
      value => {
        console.log('onResolved3()',value);
      },
      reason => {
        console.log('onRejected2()',reason);
      }
    )
  </script>
</body>
</html>